wsd/ProofKey.cpp

```diff
@@ -185,7 +185,10 @@ std::vector<unsigned char> Proof::RSA2CapiBlob(const std::vector<unsigned char>&
         throw ParseError("Proof key public exponent is longer than 4 bytes.");
     // make sure exponent length is correct; assume we are passed big-endian vectors
     std::vector<unsigned char> exponent32LE(4);
```

Comment:
Additional idea: maybe it tries to "optimize" by some memcpy-like call, taking into account not the vector size, but allocation size - and std::vector's default ctor allocates 16 elements by default (IIRC). So maybe replace the

    std::vector<unsigned char> exponent32LE(4);

with

    std::vector<unsigned char> exponent32LE;
    exponent32LE.resize(4);

so that the ctor allocates (reserves) 16 bytes, and resize makes the first 4 bytes "used"?

Reply:
Right. This is exactly what I also thought is happening. The fact that it is missing the early return means this is a bug on their side.

Another fix would be to construct `expoent32LE` with the default ctor (so we get the default allocation) and then we resize to 4 (so we make 4 bytes usable). This should work, but it's easy for someone to "optimize" this and break it again. I'll try it anyway as it's cleaner and add a comment if it works.

Reply:
The default constructor doesn't work. Instead, I use `exponent.capacity()` followed by `.resize(4)` and that works.